api/segments/models.py

```diff
@@ -84,6 +111,46 @@ def id_exists_in_rules_data(rules_data: typing.List[dict]) -> bool:
 
         return False
 
+    @hook(BEFORE_CREATE, when="version_of", is_now=None)
+    def set_default_version_to_one_if_new_segment(self):
+        if self.version is None:
+            self.version = 1
+
+    @hook(AFTER_CREATE, when="version_of", is_now=None)
+    def set_version_of_to_self_if_none(self):
+        """
+        This allows the segment model to reference all versions of
+        itself including itself.
+        """
+        self.version_of = self
+        self.save()
+
+    def deep_clone(self) -> "Segment":
+        new_segment = Segment.objects.create(
+            name=self.name,
+            description=self.description,
+            project=self.project,
+            feature=self.feature,
+            version=self.version,
+            version_of=self,
+        )
```

Comment:
Ok I'll implement that strategy. Just to double check, the timestamps will still be set to `now` for the new model, right?

Reply:
Ok this is done.

Reply:
>Just to double check, the timestamps will still be set to now for the new model, right?

Good question - I don't know. We should verify in a test. Although I guess the question is whether we want that to be the case. Since we're backfilling the history, wouldn't we _want_ e.g. the `created_at` not to change?

Reply:
Well it turns out that none of the models involved in this PR have `created_at` or `updated_at` timestamps, although they probably should, at least on the segments. Should I add them to this PR? If so, what should the default be for existing objects `None` or `now()`? Should I include conditions and rules in having timestamps as well?

Reply:
Let's add these fields to all models in the segments app, yes, but let's do it in a separate PR. Regarding the default, we _could_ be smart and look in the objects history but that might end up being a fairly mammoth migration.

Reply:
Ok sounds good I've made a new ticket for this.

https://github.com/Flagsmith/flagsmith/issues/4211

Reply:
The PR is up for the timestamps:

https://github.com/Flagsmith/flagsmith/pull/4236